boss/api/hipchat.py

```diff
@@ -0,0 +1,101 @@
+'''
+Module for hipchat API.
+'''
+
+import json
+from fabric.api import local, parallel
+from ..config import get as _get_config
+
+DEPLOYING_MESSAGE = '{user} is deploying {project_link}:{branch_link} to {server_link} server.'
+DEPLOYED_SUCCESS_MESSAGE = 'Finished deploying {project_link}:{branch_link} to {server_link} server.'
+
+
+def config():
+    ''' Get hipchat configuration. '''
+    return _get_config()['notifications']['hipchat']
+
+
+def is_enabled():
+    ''' Check if hipchat is enabled or not. '''
+    return config()['enabled']
+
+
+def create_link(url, title):
+    ''' Create a link for hipchat payload. '''
+    return '<{url}|{title}>'.format(
+        url=url,
+        title=title
+    )
+
+
+@parallel
+def notify(payload):
+    ''' Send a notification on hipchat. '''
+    command = 'curl -X POST -H "Content-type: application/json" --data \'{data}\' {url}'
+    # TODO: Don't rely on curl to do this.
+
+    local(command.format(
+        data=json.dumps(payload),
+        url=config()['base_uri'] + config()['endpoint']
```

Comment:
Let's add Hipchat API `base_uri` in the default config too, so the user won't have to put it in their config. 
Here https://github.com/kabirbaidhya/boss-cli/blob/master/boss/constants.py#L22-L25